waiter/container_delete.go

```diff
@@ -0,0 +1,73 @@
+package waiter
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/nspcc-dev/neofs-sdk-go/client"
+	apistatus "github.com/nspcc-dev/neofs-sdk-go/client/status"
+	"github.com/nspcc-dev/neofs-sdk-go/container"
+	cid "github.com/nspcc-dev/neofs-sdk-go/container/id"
+)
+
+// ContainerDeleteExecutor represents requirements to async container delete operation.
+type ContainerDeleteExecutor interface {
+	ContainerDelete(ctx context.Context, id cid.ID, prm client.PrmContainerDelete) error
+	ContainerGet(ctx context.Context, id cid.ID, prm client.PrmContainerGet) (container.Container, error)
+}
+
+// ContainerDeleteWaiter implements sync logic to container delete operation.
+type ContainerDeleteWaiter struct {
+	executor     ContainerDeleteExecutor
+	pollInterval time.Duration
+}
+
+// NewContainerDeleteWaiter is a constructor for ContainerDeleteWaiter.
+func NewContainerDeleteWaiter(executor ContainerDeleteExecutor) ContainerDeleteWaiter {
+	return ContainerDeleteWaiter{executor: executor, pollInterval: defaultPollInterval}
+}
+
+// SetPollInterval allows rewrite default poll interval.
+func (w *ContainerDeleteWaiter) SetPollInterval(interval time.Duration) {
+	w.pollInterval = interval
+}
+
+// ContainerDelete sends request to remove the NeoFS container.
+//
+// See details in [client.Client.ContainerDelete].
+func (w ContainerDeleteWaiter) ContainerDelete(ctx context.Context, id cid.ID, prm client.PrmContainerDelete) error {
+	if err := w.executor.ContainerDelete(ctx, id, prm); err != nil {
+		return nil
+	}
+
+	if err := w.isBucketDeleted(ctx, id); err != nil {
+		return nil
+	}
+
+	return nil
+}
+
+func (w ContainerDeleteWaiter) isBucketDeleted(ctx context.Context, id cid.ID) error {
+	t := time.NewTicker(w.pollInterval)
+	defer t.Stop()
+
+	var cmdGet client.PrmContainerGet
```

Comment:
cosmetic: what do you mean by "cmdGet"?

Reply:
I mean params, but `prm` looks better. Renamed